OSOE-400: Auto-generate next version number when publishing NuGet packages (#787)
Lombiq/Open-Source-Orchard-Core-Extensions: Releasing.md

﻿# Releasing a new version

> For releasing a new version of any project in this repository to NuGet there are a few handy scripts that you can use.

Reviewer: ```suggestion
We have a few handy scripts you can use to release a new NuGet package version of any project in this repository.
```

## VNext

> ⚠ This script will automatically create a tag and push it to git if successful. Make sure you are ready to release the new version before running this script.

The main script to use is `VNext.ps1`. It's a PowerShell script that you can run from the root of the repository. It will take care of everything for you given the right input.

### VNext - Parameters

- `Path`: The path to the project you are releasing e.g. `.\src\Libraries\Lombiq.HelpfulLibraries\`.
- `Type`: The type of the release. Can be `major`, `minor`, or `patch` (or can be omitted if doing a pre-release).
- `PreRelease`: Boolean flag to indicate if this is a pre-release.
- `Issue`: The issue number that this release is related to, is only used if it is a pre-release.
- `UpdateReferences`: Boolean flag to indicate if the references should be updated to this new version in the other projects in the repository.
- `Interactive`: Boolean flag to indicate if the script should ask for confirmation before pushing changes to git - defaults to true.

## VNext - Usage and examples

### Releasing a new major version

> ```shell

Reviewer: `shell` refers to Unix _.sh_ scripts. Please replace with `pwsh` across the document.

.\VNext.ps1 -path \path\to\project\ -type "major" -commitChanges $true
```

### Example release of a new major version

1. Open a PowerShell terminal.
2. Navigate to the root of the repository.
3. Run the script with the necessary parameters. For example to release a new major version of the `Lombiq.HelpfulLibraries` project and update the references in other projects, run the following command

    ```shell

>     .\VNext.ps1 -path .\src\Libraries\Lombiq.HelpfulLibraries\ -type "major" -commitChanges $true

Reviewer: Make sure all the parameter names have the correct casing, e.g. `-Path`, `-Type`, etc. (same below)
